packages/core/src/auth/activation.ts

```diff
@@ -3,16 +3,28 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import * as vscode from 'vscode'
 import { Auth } from './auth'
 import { LoginManager } from './deprecated/loginManager'
 import { fromString } from './providers/credentials'
 import { getLogger } from '../shared/logger'
-import { ExtensionUse } from './utils'
-import { isCloud9 } from '../shared/extensionUtilities'
+import { ExtensionUse, initializeCredentialsProviderManager } from './utils'
+import { isAmazonQ, isCloud9, isSageMaker } from '../shared/extensionUtilities'
 import { isInDevEnv } from '../shared/vscode/env'
 import { isWeb } from '../shared/extensionGlobals'
 
+interface SagemakerCookie {
+    authMode?: 'Sso' | 'Iam'
+}
+
 export async function initialize(loginManager: LoginManager): Promise<void> {
+    if (isAmazonQ() && isSageMaker()) {
+        // The command `sagemaker.parseCookies` is registered in VS Code Sagemaker environment.
+        const result = (await vscode.commands.executeCommand('sagemaker.parseCookies')) as SagemakerCookie
+        if (result.authMode !== 'Sso') {
+            initializeCredentialsProviderManager()
```

Comment:
how does this refactor affect toolkit ? In sagemaker studio sso mode, toolkit will still work in IAM mode.

Reply:
The higher level condition is scoped to amazonq, so it shouldn't impact aws toolkit